lib/auth/auth.go

```diff
@@ -5736,7 +5740,7 @@ func (a *Server) isMFARequired(ctx context.Context, checker services.AccessCheck
 
 // mfaAuthChallenge constructs an MFAAuthenticateChallenge for all MFA devices
 // registered by the user.
-func (a *Server) mfaAuthChallenge(ctx context.Context, user string, passwordless bool) (*proto.MFAAuthenticateChallenge, error) {
+func (a *Server) mfaAuthChallenge(ctx context.Context, user string, scope webauthnpb.ChallengeScope, allowReuse bool) (*proto.MFAAuthenticateChallenge, error) {
```

Comment:
Instead of `scope` and `allowReuse` I would pass here the (hypothetical) type that encompasses both of them.